target/target.go

```diff
@@ -98,16 +180,27 @@ func (t *Target) GetWork() *requester.Work {
 			panic(err)
 		}
 		t.Body = b.Bytes()
-		req.Header.Add("Content-Encoding", "gzip")
+		t.Config.Header.Add("Content-Encoding", "gzip")
 	}
 
-	req.ContentLength = int64(len(t.Body))
-
-	workReq := &requester.SimpleReq{
-		Request:     req,
-		RequestBody: t.Body,
-		Timeout:     t.Config.RequestTimeout,
-		QPS:         t.Config.Qps,
+	var workReq requester.Req
+	if t.Config.Stream {
+		workReq = &requester.StreamReq{
+			Method:     t.Method,
+			Url: t.Url,
+			Header: t.Config.Header,
+			Timeout: time.Duration(t.Config.RequestTimeout),
+			RunTimeout: t.Config.RunTimeout,
+			EPS: t.Config.Throttle,
+			RequestBody: [][]byte{t.Body},
```

Comment:
This would create an `[][]byte` with one entry containing the whole body. I suggest to change the `composer` already to return a `[][]byte` where every event is one entry. Otherwise the whole body is going to be sent at once in the requester, instead of one event after the other.

Reply:
great point! tried something here f7112ae44d6ea365a717a2f37ddb9d7853bc7241, did you have something like that in mind?
that complicates compression thou, any ideas?